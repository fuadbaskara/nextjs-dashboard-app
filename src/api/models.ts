export interface CommonListResponse<T> {
  result: {
    data: T[];
    count: number;
    totalCount?: number;
  } | null;
  error: unknown | Error | null;
}

export interface CommonListParams {
  id?: string;
  search?: string;
  page?: number;
  perPage?: number;
}
