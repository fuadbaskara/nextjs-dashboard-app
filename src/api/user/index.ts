// put your API calls here, you can create another file
// to make more collection of specific endpoints

import apiCall from '../axios';
import METHODS from '../methods';
import { CommonListParams, CommonListResponse } from '../models';
import { UserListResponse } from './model';

const fetchUsers = async (
  params?: CommonListParams,
): Promise<CommonListResponse<UserListResponse>> => {
  const response = await apiCall(METHODS.GET, '', params);
  return { results: response.data };
};

export default {
  fetchUsers,
};
