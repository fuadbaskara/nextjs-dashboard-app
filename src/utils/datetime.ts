import dayjs from 'dayjs';

export const DEFAULT_DATE = '0001-01-01';
export const DEFAULT_DATETIME = '0001-01-01T00:00:00Z';

export const DATE_VIEW_FORMAT = 'DD MMM YYYY';
export const COMPACT_DATE_FORMAT = 'MMMM D, YYYY';
export const COMPLETE_DATE_FORMAT = 'MMMM D, YYYY h:mm A';
export const DATETIME_VIEW_FORMAT = 'DD MMM YYYY HH:mm';
export const DATE_FORMAT_FILTER = 'DD-MM-YYYY';
export const DATE_FILTER_REPORT = 'YYYY-MM-DD';

export const dateToString = (date: string, format = COMPLETE_DATE_FORMAT): string => (date && dayjs(date).isValid() ? dayjs(date).format(format) : '');

export const parseCompleteDate = (date: string, format = COMPLETE_DATE_FORMAT): string => (date && dayjs(date).isValid() ? dayjs(date).format(format) : '');

export const dateToStringWithParseFormat = (
  date: string,
  format = DATE_VIEW_FORMAT,
  parseFormat = DATE_FILTER_REPORT,
): string => (date && dayjs(date, parseFormat).isValid() ? dayjs(date, parseFormat).format(format) : '');

export const dateTimeToString = (date: string, format = COMPLETE_DATE_FORMAT): string => (date && dayjs(date).isValid() ? dayjs(date).format(format) : '');

export const getAge = (date: string): number => dayjs().diff(date, 'year');
