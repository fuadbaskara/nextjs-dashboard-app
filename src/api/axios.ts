import axios, { AxiosPromise, AxiosRequestConfig, Method } from 'axios';

import METHODS from './methods';
import { errorInterceptor, responseInterceptor } from './interceptors';
import { toSnakeCase } from '../utils/string';

const instance = axios.create();
instance.interceptors.response.use(responseInterceptor, errorInterceptor);

const apiCall = (
  method: Method,
  endpointUrl = '',
  data: Record<string, any> = {},
  options?: AxiosRequestConfig,
  overrideBaseUrl?: string,
): AxiosPromise => {
  const API_URL = process.env.NEXT_PUBLIC_API_URL;
  const config: AxiosRequestConfig = {
    ...options,
    baseURL: overrideBaseUrl || API_URL,
    method,
    url: endpointUrl,
    headers: {
      ...(options && options.headers ? options.headers : {}),
      'Content-Type': 'application/json',
    },
  };
  const payload = { seeds: process.env.NEXT_PUBLIC_API_SEED, ...data } as Record<string, any>;
  if (method === METHODS.GET) {
    Object.keys(payload).forEach((key) => {
      if (payload[key] === null || payload[key] === '') {
        delete payload[key];
      }
    });
    config.params = toSnakeCase(payload);
  } else {
    config.data = toSnakeCase(payload);
  }
  return instance.request(config);
};

export default apiCall;
