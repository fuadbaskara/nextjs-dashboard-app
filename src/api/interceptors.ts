import { AxiosError, AxiosResponse } from 'axios';

export const responseInterceptor = <T>(response: AxiosResponse<T>) => {
  return {
    ...response,
    ...(response.data && { data: response.data }),
  };
};

export const errorInterceptor = (err: AxiosError): Promise<never> => {
  const { response } = err;
  if (response) {
    // handle interceptor here
  } else {
    throw new Error('Something wrong happened');
  }
  return Promise.reject(err);
};
