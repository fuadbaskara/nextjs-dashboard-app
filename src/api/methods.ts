import { Method } from 'axios';

const METHODS = {
  GET: 'GET' as Method,
  POST: 'POST' as Method,
  PUT: 'PUT' as Method,
  PATCH: 'PATCH' as Method,
  DELETE: 'DELETE' as Method,
};

export default METHODS;
