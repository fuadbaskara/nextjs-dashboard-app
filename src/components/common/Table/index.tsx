/* eslint-disable object-curly-newline */
import React, { useEffect, useState } from 'react';
import { AiFillCaretDown, AiFillCaretUp } from 'react-icons/ai';
import clsx from 'clsx';
import { uuid } from '@/utils/generator';
import Pagination from '../Pagination';
import { useAppSelector } from '@/hooks';
import { filterSelector } from '@/store/slices/filter';
import { SortOrder } from '@/api/models';

export type TableProps<T> = {
  title?: string;
  dataSource: T[];
  rowKey?: string;
  pagination: {
    current: number;
    onChange: (page: number) => void;
  };
  columns: {
    title: string;
    dataIndex: string;
    key?: string | number;
    sorter?: (orderBy: string, sortOrder: SortOrder) => void;
    render: (record: T) => React.ReactNode | string | null;
  }[];
  loading: boolean;
};

function Table<T>({ title, dataSource, rowKey, pagination, columns, loading }: TableProps<T>) {
  const { filter } = useAppSelector(filterSelector);
  const [sort, setSort] = useState({ by: '', order: '' });

  useEffect(() => {
    if (!filter.sortBy && !filter.sortOrder) setSort({ by: '', order: '' });
  }, [filter.sortBy, filter.sortOrder]);

  return (
    <div className="flex flex-col">
      <div className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8">
        <div className="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8">
          <div className="overflow-hidden border-b border-gray-200 shadow sm:rounded-lg">
            {title && <h2>{title}</h2>}
            <div id="table-container">
              <table className="min-w-full divide-y divide-neutral-400">
                <thead className="bg-blue-500">
                  <tr>
                    {(columns || []).map((col) => (
                      <th
                        className="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                        key={uuid()}
                      >
                        <div className="flex items-center justify-between">
                          <span>{col.title}</span>
                          {col.sorter && (
                            <span>
                              <AiFillCaretUp
                                className={clsx(
                                  sort.by === col.dataIndex && sort.order === 'ascend'
                                    ? 'text-blue-900'
                                    : '',
                                  'hover:cursor-pointer',
                                )}
                                onClick={() => {
                                  if (col.sorter) {
                                    setSort({
                                      by: col.dataIndex,
                                      order: 'ascend',
                                    });
                                    col.sorter(col.dataIndex, 'ascend');
                                  }
                                }}
                              />
                              <AiFillCaretDown
                                className={clsx(
                                  sort.by === col.dataIndex && sort.order === 'descend'
                                    ? 'text-blue-900'
                                    : '',
                                  'hover:cursor-pointer',
                                )}
                                onClick={() => {
                                  if (col.sorter) {
                                    setSort({
                                      by: col.dataIndex,
                                      order: 'descend',
                                    });
                                    col.sorter(col.dataIndex, 'descend');
                                  }
                                }}
                              />
                            </span>
                          )}
                        </div>
                      </th>
                    ))}
                  </tr>
                </thead>
                {!loading && (
                  <tbody>
                    {(dataSource || []).map((data: T | any, idx: number) => (
                      <tr
                        key={data[`${rowKey}`] || uuid()}
                        className={`${idx % 2 === 0 ? 'bg-white' : 'bg-cyan-50'} hover:bg-cyan-100`}
                      >
                        {(columns || []).map((col) => (
                          <td
                            key={col.key || col.dataIndex}
                            className="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                          >
                            {col.render(data)}
                          </td>
                        ))}
                      </tr>
                    ))}
                  </tbody>
                )}
              </table>
              {loading && (
                <div className="flex h-80 w-full items-center justify-center">
                  <span className="text-xl">Loading...</span>
                </div>
              )}
            </div>
          </div>
        </div>
      </div>
      <div className="mt-4 flex justify-end">
        <Pagination current={pagination.current} onChange={pagination.onChange} />
      </div>
    </div>
  );
}

Table.defaultProps = {
  title: '',
  rowKey: undefined,
};

export default Table;
