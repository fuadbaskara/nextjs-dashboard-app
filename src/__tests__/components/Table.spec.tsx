/* eslint-disable testing-library/no-node-access */
import React from 'react';
import { render } from '../test-utils';
import Table from '@/components/common/Table';
import { SortOrder } from '@/api/models';
import { dateToString } from '@/utils/datetime';
import { mockUser } from '../../__mocks__/mock-data/user-list';
import { User } from '@/api/user/model';

const userList: User[] = mockUser.results;

type Props = {
  onSort: (sortBy: string, sortOrder: SortOrder) => void;
};

const onSorting = (sortBy: string, sortOrder: SortOrder) => {
  jest.fn(() => {
    return { sortBy, sortOrder };
  });
};

const columns = ({ onSort }: Props) => {
  return [
    {
      title: 'Username',
      dataIndex: 'username',
      key: 'username',
      render: (record: User) => <div>{record.login.username || '-'}</div>,
    },
    {
      title: 'Name',
      dataIndex: 'name',
      key: 'name',
      render: (record: User) => (
        <div>{`${record.name.first} ${record.name.last}` || '-'}</div>
      ),
      sorter: (orderBy: string, sortOrder: SortOrder) =>
        onSort(orderBy, sortOrder),
    },
    {
      title: 'Email',
      dataIndex: 'email',
      key: 'email',
      render: (record: User) => <div>{record.email || '-'}</div>,
      sorter: (orderBy: string, sortOrder: SortOrder) =>
        onSort(orderBy, sortOrder),
    },
    {
      title: 'Gender',
      dataIndex: 'gender',
      key: 'gender',
      render: (record: User) => <div>{record.gender || '-'}</div>,
      sorter: (orderBy: string, sortOrder: SortOrder) =>
        onSort(orderBy, sortOrder),
    },
    {
      title: 'Registered Date',
      dataIndex: 'date',
      key: 'date',
      render: (record: User) => (
        <div>
          {record.registered.date
            ? dateToString(`${record.registered.date}`)
            : '-'}
        </div>
      ),
      sorter: (orderBy: string, sortOrder: SortOrder) =>
        onSort(orderBy, sortOrder),
    },
  ];
};

const renderTable = () =>
  render(
    <Table
      dataSource={userList}
      columns={columns({ onSort: onSorting })}
      loading={false}
      rowKey="cell"
      pagination={{
        current: 1,
        onChange: (currPage) => jest.fn(() => currPage),
      }}
    />,
  );

describe('Table Component', () => {
  test('renders Table correctly with props:', () => {
    renderTable();
    const tableRows = document.querySelectorAll('tr');
    expect(tableRows).toHaveLength(11);

    const { asFragment } = renderTable();
    expect(asFragment()).toMatchSnapshot();
  });
});
