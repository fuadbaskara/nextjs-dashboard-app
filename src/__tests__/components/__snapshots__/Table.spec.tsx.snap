// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component renders Table correctly with props: 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col"
  >
    <div
      class="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
    >
      <div
        class="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8"
      >
        <div
          class="overflow-hidden border-b border-gray-200 shadow sm:rounded-lg"
        >
          <div
            id="table-container"
          >
            <table
              class="min-w-full divide-y divide-neutral-400"
            >
              <thead
                class="bg-blue-500"
              >
                <tr>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Username
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Name
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Email
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Gender
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                  <th
                    class="px-6 py-6 text-left text-sm font-medium tracking-wider text-white"
                  >
                    <div
                      class="flex items-center justify-between"
                    >
                      <span>
                        Registered Date
                      </span>
                      <span>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                          />
                        </svg>
                        <svg
                          class="hover:cursor-pointer"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  class="bg-white hover:bg-cyan-100"
                >
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      organicelephant410
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      Hailey Taylor
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      hailey.taylor@example.com
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      female
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      22-10-2003 11:59
                    </div>
                  </td>
                </tr>
                <tr
                  class="bg-cyan-50 hover:bg-cyan-100"
                >
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      smallbear142
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      Hector Bennett
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      hector.bennett@example.com
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      male
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      02-03-2020 14:58
                    </div>
                  </td>
                </tr>
                <tr
                  class="bg-white hover:bg-cyan-100"
                >
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      greenladybug666
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      Dorian Girard
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      dorian.girard@example.com
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      male
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      03-11-2006 15:07
                    </div>
                  </td>
                </tr>
                <tr
                  class="bg-cyan-50 hover:bg-cyan-100"
                >
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      crazydog363
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      Maxine Gonzales
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      maxine.gonzales@example.com
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      female
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      10-01-2004 17:06
                    </div>
                  </td>
                </tr>
                <tr
                  class="bg-white hover:bg-cyan-100"
                >
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      whitebird714
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      Noe Roy
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      noe.roy@example.com
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      male
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      12-04-2015 13:17
                    </div>
                  </td>
                </tr>
                <tr
                  class="bg-cyan-50 hover:bg-cyan-100"
                >
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      whitefish690
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      Lucy Russell
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      lucy.russell@example.com
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      female
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      12-06-2015 06:35
                    </div>
                  </td>
                </tr>
                <tr
                  class="bg-white hover:bg-cyan-100"
                >
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      redleopard421
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      Shannon Taylor
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      shannon.taylor@example.com
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      female
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      04-04-2021 09:35
                    </div>
                  </td>
                </tr>
                <tr
                  class="bg-cyan-50 hover:bg-cyan-100"
                >
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      orangepanda607
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      Enrique Robinson
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      enrique.robinson@example.com
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      male
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      08-01-2017 16:07
                    </div>
                  </td>
                </tr>
                <tr
                  class="bg-white hover:bg-cyan-100"
                >
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      blackbutterfly179
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      Lily Rodriguez
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      lily.rodriguez@example.com
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      female
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      06-09-2021 06:41
                    </div>
                  </td>
                </tr>
                <tr
                  class="bg-cyan-50 hover:bg-cyan-100"
                >
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      tinygorilla272
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      Titouan Vincent
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      titouan.vincent@example.com
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      male
                    </div>
                  </td>
                  <td
                    class="font-small whitespace-nowrap px-6 py-4 text-sm text-gray-500"
                  >
                    <div>
                      17-04-2008 14:18
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-4 flex justify-end"
    >
      <ul
        class="flex"
      >
        <li>
          <button
            class="inline-flex items-center rounded-l-md border border-gray-300 bg-white px-2 py-2 text-sm font-medium text-blue-500 hover:bg-blue-50"
            data-testid="prev-page-btn"
            style="font-size: 1.03rem;"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            class="z-10 border-blue-500 bg-blue-100 text-blue-600 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            data-testid="page-1-btn"
            type="button"
          >
            1
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            data-testid="page-2-btn"
            type="button"
          >
            2
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            data-testid="page-3-btn"
            type="button"
          >
            3
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            data-testid="page-4-btn"
            type="button"
          >
            4
          </button>
        </li>
        <li>
          <button
            class="border-gray-300 bg-white text-blue-500 hover:bg-blue-50 inline-flex items-center border px-4 py-1.5 text-sm font-medium"
            data-testid="page-5-btn"
            type="button"
          >
            5
          </button>
        </li>
        <li>
          <button
            class="inline-flex items-center rounded-r-md border border-gray-300 bg-white px-2 py-2 text-sm font-medium text-blue-500 hover:bg-blue-50"
            data-testid="next-page-btn"
            style="font-size: 1.03rem;"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
